dotnet/src/dotnetcore/Providers/Messaging/GXQueue/Queue.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using GeneXus.Utils;
+
+namespace GeneXus.Messaging.Common
+{
+	public interface IQueue
+	{
+		public const int MAX_NUMBER_MESSAGES = 10;
+		public const bool DELETE_CONSUMED_MESSAGES = false;
+		public const int WAIT_TIMEOUT = 10;
+		public const int VISIBILITY_TIMEOUT = 60;
+
+		int GetQueueLength(out bool success);
+		MessageQueueResult SendMessage(SimpleQueueMessage simpleQueueMessage, out bool success);
+		MessageQueueResult SendMessage(SimpleQueueMessage simpleQueueMessage, MessageQueueOptions messageQueueOptions, out bool success);
+		IList<MessageQueueResult> SendMessages(IList<SimpleQueueMessage> simpleQueueMessages, MessageQueueOptions messageQueueOptions, out bool success);
+		IList<SimpleQueueMessage> GetMessages(out bool success);
+		IList<SimpleQueueMessage> GetMessages(MessageQueueOptions messageQueueOptions, out bool success);
+		MessageQueueResult DeleteMessage(out bool success);
```

Comment:
Why this method? In the Spec the DeleteMessage receives a MessageId. 

This method should be called "Dequeue". It means take the first and remove it.

Reply:
Yes, it isn't in the spec, but I found it usefull to dequeue. I can rename it, ok?

Reply:
ok!